src/array/cpu/labor_pick.h

```diff
@@ -0,0 +1,347 @@
+/*!
+ *   Copyright (c) 2022, NVIDIA Corporation
+ *   Copyright (c) 2022, GT-TDAlab (Muhammed Fatih Balin & Umit V. Catalyurek)
+ *   All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ *
+ * \file array/cpu/labor_pick.h
+ * \brief Template implementation for layerwise pick operators.
+ */
+
+#ifndef DGL_ARRAY_CPU_LABOR_PICK_H_
+#define DGL_ARRAY_CPU_LABOR_PICK_H_
+
+#include <dgl/random.h>
+#include <dgl/array.h>
+#include <dmlc/omp.h>
+#include <dgl/runtime/parallel_for.h>
+#include <parallel_hashmap/phmap.h>
+#include <functional>
+#include <algorithm>
+#include <string>
+#include <vector>
+#include <numeric>
+#include <memory>
+#include <utility>
+#include <cmath>
+#include <iostream>
+#include <pcg_random.hpp>
+
+namespace dgl {
+namespace aten {
+namespace impl {
+
+// Template for picking non-zero values row-wise.
+template <typename IdxType, typename FloatType>
+std::pair<COOMatrix, FloatArray> CSRLaborPick(
+    CSRMatrix mat,
+    IdArray NIDs,
+    IdArray rows,
+    int64_t __num_picks,
+    FloatArray prob,
+    IdArray random_seed,
+    IdArray cnt,
+    int importance_sampling) {
+  using namespace aten;
+  const IdxType* indptr = static_cast<IdxType*>(mat.indptr->data);
+  const IdxType* indices = static_cast<IdxType*>(mat.indices->data);
+  const IdxType* data = CSRHasData(mat)? static_cast<IdxType*>(mat.data->data) : nullptr;
+  const IdxType* rows_data = static_cast<IdxType*>(rows->data);
+  const IdxType* nids = NIDs->shape[0] ? static_cast<IdxType*>(NIDs->data) : nullptr;
+  const auto num_rows = rows->shape[0];
+  const IdxType num_picks = __num_picks;
```

Comment:
Any problem with using `int64_t`?

Reply:
No I mean why not use `int64_t` instead of `IdxType` for `num_picks`.

Reply:
Because num_picks is compared and used with variables of type IdxType, so when I call a function like std::min, then if both arguments are of type IdxType, no casting is needed.

Reply:
But it turns out that kind of thing happens on a single line so I will just use int64_t